### PR TITLE
Added jsdocs for grid component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
#131

### CODE REVIEW

File: src/app/public/modules/grid/grid-column.component.ts [hunk 1]
```diff
@@ -26,63 +26,118 @@ import {
...
+  public locked: boolean;
 
-  /* tslint:disable:no-input-rename */
+  /**
+   * Specifies a search function to apply for the specific column. By default,
+   * the column executes a string compare on the column data.
+   */
+  /* tslint:disable-next-line:no-input-rename */
   @Input('search')
   public searchFunction: (value: any, searchText: string) => boolean = this.search;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/grid/grid-column.component.ts:116]: @Blackbaud-SteveBrush This shows a default as `this.search`. Is there any way we can hide that?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-jsdocs](https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-jsdocs)

File: src/app/public/modules/grid/grid-column.component.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/grid/grid-column.component.ts:116]: You can provide a custom `@default` value. Would that work?

File: src/app/public/modules/grid/types/grid-message.ts [hunk 2]
```diff
@@ -3,7 +3,15 @@ import {
...
 
 export interface SkyGridMessage {
+
+  /**
+   * Indicates what type of message is being sent.
+   */
   type: SkyGridMessageType;
+
+  /**
+   * @internal
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/types/grid-message.ts:13]: This should be documented I believe? @Blackbaud-TrevorBurch

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/grid/types/grid-message.ts:13]: Same with all the `grid-row-delete-*` files below.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/types/grid-message.ts:13]: All the delete logic is handled internally. The consumer shouldn't need to access the `data` object here or the delete types.

File: src/app/public/modules/grid/grid-column.component.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid-column.component.ts:117]: Yep that worked. I kinda paraphrased the function a bit to find a nice balance of communicating what its doing and being concise.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-jsdocs](https://developer.blackbaud.com/skyux-grids?_sp.spa=pr-jsdocs)

**Pull request merged**